airflow/providers/amazon/aws/hooks/glue.py

```diff
@@ -0,0 +1,214 @@
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+import os.path
+import time
+
+
+class AwsGlueJobHook(AwsBaseHook):
+    """
+    Interact with AWS Glue - create job, trigger, crawler
+
+    :param job_name: unique job name per AWS account
+    :type job_name: str
+    :param desc: job description
+    :type desc: str
+    :param concurrent_run_limit: The maximum number of concurrent runs allowed for a job
+    :type concurrent_run_limit: int
+    :param script_location: path to etl script either on s3 or local
+    :type script_location: str
+    :param conns: A list of connections used by the job
+    :type conns: list
+    :param retry_limit: Maximum number of times to retry this job if it fails
+    :type retry_limit: int
+    :param num_of_dpus: Number of AWS Glue DPUs to allocate to this Job
+    :type num_of_dpus: int
+    :param region_name: aws region name (example: us-east-1)
+    :type region_name: str
+    :param iam_role_name: AWS IAM Role for Glue Job
+    :type iam_role_name: str
+    :param s3_bucket: S3 bucket where logs and local etl script will be uploaded
+    :type s3_bucket: str
+    """
+
+    def __init__(self,
+                 job_name=None,
+                 desc=None,
+                 concurrent_run_limit=1,
+                 script_location=None,
+                 conns=None,
+                 retry_limit=0,
+                 num_of_dpus=10,
+                 aws_conn_id='aws_default',
+                 region_name=None,
+                 iam_role_name=None,
+                 s3_bucket=None, *args, **kwargs):
+        self.job_name = job_name
+        self.desc = desc
+        self.concurrent_run_limit = concurrent_run_limit
+        self.script_location = script_location
+        self.conns = conns or ["s3"]
+        self.retry_limit = retry_limit
+        self.num_of_dpus = num_of_dpus
+        self.aws_conn_id = aws_conn_id
+        self.region_name = region_name
+        self.s3_bucket = s3_bucket
+        self.role_name = iam_role_name
+        self.S3_PROTOCOL = "s3://"
+        self.S3_ARTIFACTS_PREFIX = 'artifacts/glue-scripts/'
+        self.S3_GLUE_LOGS = 'logs/glue-logs/'
+        super(AwsGlueJobHook, self).__init__(*args, **kwargs)
+
+    def get_conn(self):
+        conn = self.get_client_type('glue', self.region_name)
+        return conn
+
+    def list_jobs(self):
+        conn = self.get_conn()
+        return conn.get_jobs()
+
+    def get_iam_execution_role(self):
+        """
+        :return: iam role for job execution
+        """
+        iam_client = self.get_client_type('iam', self.region_name)
+
+        try:
+            glue_execution_role = iam_client.get_role(RoleName=self.role_name)
+            self.log.info("Iam Role Name: %s", self.role_name)
+            return glue_execution_role
+        except Exception as general_error:
+            self.log.error(f'Failed to create aws glue job, error: {str(general_error)}')
+            raise
```

Comment:
Can you check if you can reuse some code of `AwsBaseHook`? If you search for `iam` you can see that there is iam auth already implemented maybe you can use this. WDYT?

Reply:
Hey, Thanks for pointing out, It is using get_client_type and get_role from awsbasehook actually, so its already taking benefit from that implementation

Reply:
Yes but the `get_role` is accessing the `boto` library directly which should actually only do the related hook. If we had one it would be `AwsIamHook`.

But if you cannot make it work with the functions in `AwsBaseHook` you can access it for now like you do.

Reply:
"But if you cannot make it work with the functions in AwsBaseHook you can access it for now like you do." with boto object. as it is right now?

Reply:
Yes.